### PR TITLE
Filter reviewer dashboard tables by ReviewerSettings

### LINKED ISSUE #2823: The "Your previous reviews" table is not filters after what users can access
I suggest we continue to list all the users reviews but the ones they can not access are not links. Maybe greyed out? With a tooltip that explains that they no longer have access to it?

### PR DESCRIPTION
**Pull request by frjo**
Fixes #2823

### CODE REVIEW

File: hypha/apply/dashboard/views.py [hunk 1]
```diff
@@ -126,6 +130,15 @@ def projects_to_approve(self):
...
 
     def my_reviewed(self, submissions):
+        '''
+        If use_settings variable is set for ReviewerSettings use settings
+        parameters to filter submissions or return only reviewed_by as it
+        was by default.
+        '''
+        reviewer_settings = ReviewerSettings.for_request(self.request)
+        if reviewer_settings.use_settings:
+            submissions = submissions.for_reviewer_settings(self.request.user, reviewer_settings)
```
Comment (sks444) [inline hypha/apply/dashboard/views.py]: @frjo, this is AdminDashboardView this should be done only for [ReviewerDashboardView](https://github.com/HyphaApp/hypha/blob/4b865bb04106ab7a409fdf0dc3d09b75e7602c45/hypha/apply/dashboard/views.py#L175)

Comment (frjo) [reply hypha/apply/dashboard/views.py]: Oops, fixed now.

Comment (frjo) [reply hypha/apply/dashboard/views.py]: But you think the method is ok otherwise?

Comment (sks444) [reply hypha/apply/dashboard/views.py]: Yeah. This looks good. I'm just wondering if this should also be done for [awaiting_reviews](https://github.com/HyphaApp/hypha/blob/4b865bb04106ab7a409fdf0dc3d09b75e7602c45/hypha/apply/dashboard/views.py#L201) and my_flagged.

Comment (frjo) [reply hypha/apply/dashboard/views.py]: Good point!

**Review by sks444 — APPROVED** [review]

**Pull request merged**